tests/utils/test_log_handlers.py

```diff
@@ -754,3 +754,15 @@ def test_permissions_for_new_directories():
             os.umask(old_umask)
     finally:
         shutil.rmtree(tmp_path)
+
+
+def test_permissions_for_home_directory_subdir():
+    home_dir = Path.home()
+    for dir in home_dir.glob("*"):
+        if dir.is_dir():
+            with patch("pathlib.Path.chmod") as mock_chmod:
+                _change_directory_permissions_up(dir, 0o000)
+                # Should only be called once - for the subdir - but not for the home directory
+                mock_chmod.assert_called_once_with(0o000)
+            return
+    pytest.fail("No directory found in home directory")
```

Comment:
Hm... I think we need specific marker which allow run such tests tests only in docker/breeze.
One minor changes: test or tested function and it might do some dangerous action into the developers machine

Reply:
Something similar we have into impersonation tests:
https://github.com/apache/airflow/blob/be2c93a92b2ce5f05cc94715f8621385cd17e284/tests/core/test_impersonation_tests.py#L92-L101

Reply:
Hm... I think we need specific marker

> I believe we had one in the past - but can't find it easily

Reply:
Another option would be use [`monkeypatch`](https://docs.pytest.org/en/latest/how-to/monkeypatch.html#monkeypatching-functions), I think it already have home patching but unfortunetly it doesn't

```python
from pathlib import Path

import pytest


@pytest.fixture
def fake_home(tmp_path_factory, monkeypatch) -> Path:
    home = tmp_path_factory.mktemp("home")

    def mockreturn():
        return home

    monkeypatch.setattr(Path, "home", mockreturn)
    return home

def test_foo_bar(fake_home):
    assert fake_home == Path.home()
    
    from os.path import expanduser
    assert fake_home.as_posix() != expanduser("~")
```

Reply:
> I believe we had one in the past - but can't find it easily

Yeah I've also look that recently, without any luck.

Reply:
How about just checking for existence of subdir ".ssh" and if so - stop?

Reply:
Added the marker.